### DISCUSSION TITLE
Zenity port, would it be an acceptable formula?

### DISCUSSION THREAD

**Discussion by ncruces** [Homebrew/discussions#1630] [Tap maintenance and Homebrew development]
Hi all,

I'm working on a [Zenity](https://help.gnome.org/users/zenity/stable/) *"port"* to macOS and Windows. The idea is that it provides a command that's (as much as possible) compatible with `zenity`, but doesn't require X11 and looks native on both platforms.

It's on GitHub: [github.com/ncruces/zenity](https://github.com/ncruces/zenity). This [issue](https://github.com/ncruces/zenity/issues/11) tracks the current status of the port, and you can find screenshots in the [wiki](https://github.com/ncruces/zenity/wiki).

I maintain a [tap](https://github.com/ncruces/homebrew-tap) for it, and you can install it on macOS and WSL with:

    brew install ncruces/tap/zenity

The WSL version cross-compiles and uses the Windows version.

I'd like to understand if this is something that might be an [acceptable formula](https://docs.brew.sh/Acceptable-Formulae) at `homebrew/core`. I'd have to call it something else (`zenity-port`, `zenity-native`... something else?), but to provide "compatibility" the port would conflict with the `zenity` [formula](https://formulae.brew.sh/formula/zenity) (build a same named binary). 

The biggest advantage to users, is that I can't really provide bottles on all platforms without taking advantage of your fantastic CI (and thank you for that!)

If there's interest I can pursue this, but since self-submitting is [frowned upon](https://docs.brew.sh/Acceptable-Formulae#niche-or-self-submitted-stuff), I obviously won't push it if there's little to no interest.

Thanks!

**Comment by SMillerDev** [comment]
Regardless of the status in homebrew/core, you can add bottles using GitHub Actions: https://brew.sh/2020/11/18/homebrew-tap-with-bottles-uploaded-to-github-releases/

> If there's interest I can pursue this, but since self-submitting is frowned upon, I obviously won't push it if there's little to no interest.

This rule is mostly to avoid people adding every weekend project to Homebrew. The popularity metrics serve a similar purpose for GitHub repos. If your software meets the other criteria it's fine to create an incidental PR for your own software.

**Reply by ncruces** [reply]
Thanks! I couldn't get GitHub Actions working the last time I've tried, especially on Linux (because it's only supposed to work on WSL), but I'll try again. Would GitHub Actions help me build a bootle for say `arm64_big_sur`?

If so, I imagine the custom tap is fine for my users. Having to download Go to build from source is (currently) the big downside relative to formulæ on core.

**Reply by ncruces** [reply]
I've fixed my issues with GitHub Actions.

There's an [issue](https://github.com/Homebrew/brew/pull/11507) where the default actions fail on Linux if you add a cask.
And I needed some improvements to get my formula to build on Linux CI, since it's supposed to be WSL only.

This is nice, since it makes bottling easier, but it only builds bottles for Linux (in this case, WSL) and macOS Catalina. I have to add Big Sur manually, and there's no ARM support.

So I still think it would be useful to users to have it added to the core taps?

**Reply by SMillerDev** [reply]
The catalina bottles will likely work with newer macOS versions too. But ARM would be the real pain point.

**Reply by ncruces** [reply]
I though so myself, but they didn't on Big Sur (that's what I'm on), which surprised me.
Anyhow, this helps, and I can add those manually (WSL was a bigger pain, and this solved it, so thanks!)
